index.js

```diff
@@ -6,25 +6,25 @@ module.exports = {
   /**
    * Get information about the OpenSSL version installed where this module is run.
    * Returns a promise.
-   * @returns {promise}
+   * @returns {Promise<{ ciphers: { supported: string[], unsupported: string[] }, protocols: { supported: string[], unsupported: string[] }, version: string }>}
    */
   getOpenSSLCapabilities: function() {
     return opensslinfo.getOpenSSLCapabilities();
   },
   /**
    * Determine the supported SSL/TLS protocols and ciphers for a server.
    * Returns a promise.
-   * @param hostData {{ host: string, port: number, [servername]: string }}
-   * @returns {promise}
+   * @param hostData {{ host: string, port: number, servername?: string, rejectUnauthorized?: boolean, minDHSize?: number }}
+   * @returns {Promise<{ cert: { altNames: string[], extensions: Object, fingerPrint: string, issuer: Object, notAfter: Date, notBefore: Date, publicKey: Object, serial: string, signatureAlgorithm; string, version: number }, certPem: string, ciphers: { TLSv1_1_method?: { disabled: string[], enabled: string[], name: string, unsupported: string[] }, TLSv1_2_method?: { disabled: string[], enabled: string[], name: string, unsupported: string[] }, TLSv1_method?: { disabled: string[], enabled: string[], name: string, unsupported: string[] }, SSLv3_method?: { disabled: string[], enabled: string[], name: string, unsupported: string[] }, SSLv2_method?: { disabled: string[], enabled: string[], name: string, unsupported: string[] }}, host: string, port: number, protocols: Object[] }>}
```

Comment:
This object definition is large and cumbersome, and not complete. I feel like parameters could be described as detailed in http://usejsdoc.org/tags-param.html#parameters-with-properties if there is a real need to call out all of the properties.

Reply:
That is a good idea.
There is also the posibillity to use TypeScript type definition files to document and decouple code doc's from the code itself.

What do you say?

Reply:
This could be re-used in parts [DefinitelyTyped/types/node/v9/index.d.ts#L4836](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0897921174860ec3d5318992d2323b3ae8100a68/types/node/v9/index.d.ts#L4836)

Reply:
I'm not against adding TS typing files.